Gems/EMotionFX/Code/Source/Integration/Components/ApplyMotionSetComponent.cpp

```diff
@@ -0,0 +1,187 @@
+#if defined(CARBONATED)
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+
+#include <MCore/Source/AttributeString.h>
+#include <EMotionFX/Source/AnimGraph.h>
+
+#include <Integration/Components/ApplyMotionSetComponent.h>
+#include <Integration/AnimGraphComponentBus.h>
+#include <EMotionFX/Source/AnimGraphInstance.h>
+#include <AzCore/Asset/AssetSerializer.h>
+
+namespace EMotionFX
+{
+    namespace Integration
+    {
+        //////////////////////////////////////////////////////////////////////////
+        void ApplyMotionSetComponent::Configuration::Reflect(AZ::ReflectContext* context)
+        {
+            auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+            if (serializeContext)
+            {
+                serializeContext->Class<Configuration>()
+#if defined(CARBONATED)

```

Comment:
Looks like inner #ifdef/#endif for CARBONATED are not necessary 
here and below
due to we use #if defined(CARBONATED) for whole file in the line 1
?